### DISCUSSION TITLE
More discriminated picker for `oneOf` types

### DISCUSSION THREAD

**Discussion by tpoll** [scalar/scalar#1202] [Ideas]
I have an openApi spec that looks like https://gist.githubusercontent.com/tpoll/c1894e88dd5f55d513f098a8f0d368c2/raw/486799140029cf62cfca61fb96a5185204d93baa/spec.json

It includes a `oneOf` field with a discriminated value mapping.

In the scalar demo it looks like: 
<img width="536" alt="Screen Shot 2024-03-15 at 12 04 24 PM" src="https://github.com/scalar/scalar/assets/4765779/f0a3362c-ee53-48c0-a093-72c41ef07068">

In redocly demo it looks like:
<img width="1447" alt="Screen Shot 2024-03-15 at 12 03 01 PM" src="https://github.com/scalar/scalar/assets/4765779/6820680f-1ed3-42bc-a844-113e797931a5">

selecting the different `oneOf` type then show the specific fields that are relevant.

I personally find the redocly UI easier to understand, so it would be cool if scalar could support that style.

**Comment by marclave** [comment]
appreciate the discussion @tpoll !

I agree that having a selector makes it easier to consume!

@hanspagel @cameronrohani what do you think about this? I wonder if we can somehow accommodate both options 🤔
